solr/core/src/java/org/apache/solr/core/CoreDescriptor.java

```diff
@@ -182,7 +182,7 @@ public CoreDescriptor(String coreName, CoreDescriptor other) {
    */
   public CoreDescriptor(String name, Path instanceDir, Map<String, String> coreProps,
                         Properties containerProperties, ZkController zkController) {
-    this.instanceDir = instanceDir;
+    this.instanceDir = instanceDir.toAbsolutePath();
```

Comment:
A bit late, but I don't think this is necessary, as all callers will send absolute paths. And if you ever get a relative path, resolving it with `toAbsolutePath()` leads to it being relative to whatever CWD the app is started with, while the typical resolving of relative `instanceDir` is to resolve it relative to CoreContainer#coreRootDirectory.

Reply:
That makes sense; thanks.  I see that all callers send absolute paths already.  So I think I should change this to throw an exception if it isn't absolute to protect us from mistakes.

I'll push a new commit here for review.  I've never done that for an already merged PR; we'll see how it goes.